collector/textfile.go

```diff
@@ -271,10 +301,20 @@ fileLoop:
 		// a failure does not appear fresh.
 		mtimes[f.Name()] = f.ModTime()
 
-		for _, mf := range parsedFamilies {
+		for _, metricFamily := range parsedFamilies {
+			metricFamilies = append(metricFamilies, metricFamily)
+		}
+	}
+
+	if duplicateMetricEntry(metricFamilies) {
+		log.Errorf("Duplicate metrics detected in files")
+		error += 1.0
```

Comment:
Hm, this would mean we could end up with `windows_textfile_scrape_error == 2`, if there are any other errors? Is there a reason to not just set it to 1 instead?

Reply:
Yes, no reason to add here. I changed this to `=` rather than the current `+=`.